pulp-glue/pulp_glue/common/authentication.py

```diff
@@ -78,10 +81,12 @@ def retrieve_token(self) -> None:
         data = {
             "client_id": self.client_id,
             "client_secret": self.client_secret,
-            "scope": " ".join(self.scopes),
             "grant_type": "client_credentials",
         }
 
+        if scope := " ".join(self.scopes):
+            data["scope"] = scope
```

Comment:
Some Idps doesn't require a scope parameter to issue a token. And since it's not required, maybe sending it as an empty value could cause an issue with the IdP itself.

Reply:
https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2

```
scope
         OPTIONAL.  The scope of the access request as described by
         [Section 3.3](https://datatracker.ietf.org/doc/html/rfc6749#section-3.
```

Reply:
This tells me it's optional. But is it allowed to send an empty string? If yes, does it convey a different semantic?

Reply:
As we stated, if the scope is to be used, it must not be empty.